crates/bevy_app/src/main_schedule.rs

```diff
@@ -0,0 +1,127 @@
+use crate::App;
+use bevy_ecs::{
+    schedule::ScheduleLabel,
+    system::{Local, Resource},
+    world::{Mut, World},
+};
+
+/// The schedule that contains the app logic that is evaluated each tick of [`App::update()`].
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Main;
+
+/// The schedule that runs before [`Startup`].
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct PreStartup;
+
+/// The schedule that runs once when the app starts.
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Startup;
+
+/// The schedule that runs once after [`Startup`].
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct PostStartup;
+
+/// The schedule that runs once when the app starts.
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct First;
+
+/// The schedule that contains work required to make [`Update`] logic.
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct PreUpdate;
+
+/// Runs state transitions.
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct StateTransition;
+
+/// Runs the [`FixedUpdate`] schedule in a loop according until all relevant elapsed time has been "consumed".
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct FixedUpdateLoop;
+
+/// The schedule that contains systems which only run after a fixed period of time has elapsed.
+///
+/// The exclusive `run_fixed_update_schedule` system runs this schedule.
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct FixedUpdate;
+
+/// The schedule that contains app logic.
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Update;
+
+/// The schedule that contains systems that respond to [`Update`].
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct PostUpdate;
+
+/// Runs last in the schedule
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Last;
+
+/// Defines the schedules to be run for the [`Main`] schedule, including
+/// their order.
+#[derive(Resource, Debug)]
+pub struct MainScheduleOrder {
+    labels: Vec<Box<dyn ScheduleLabel>>,
```

Comment:
I think this would actually be quite valuable to make `pub` for advanced users (and system visualizers).

Reply:
This at least needs a read-only accessor.